### PR TITLE
Delayed message exchange bug

### PR DESCRIPTION
**Pull request by snichme**

### CODE REVIEW

File: src/avalanchemq/vhost.cr [hunk 1]
```diff
@@ -40,8 +40,7 @@ module AvalancheMQ
     @connections = Array(Client).new(512)
     @segments : Hash(UInt32, MFile)
     EXCHANGE_TYPES = %w(direct fanout topic headers
-      x-federation-upstream x-delayed-message
```
Comment (baelter) [inline src/avalanchemq/vhost.cr:43]: This makes it incompatible with rabbitmq, no?

Comment (snichme) [reply src/avalanchemq/vhost.cr:43]: No, that list only controls which types are listen in the UI. The API looks the same. 

In the UI you instead select which exchange you want and check "Delayed"  checkbox

File: src/avalanchemq/queue/queue.cr [hunk 2]
```diff
@@ -536,19 +536,18 @@ module AvalancheMQ
...
-          if expire_in > 0
-            @log.debug { "No more message to expire" }
-            break
-          end
-
-          expire_msg(sp, :expired)
-          if (i += 1) == 8192
-            Fiber.yield
-            i = 0
+          if expire_in < 0
```
Comment (carlhoerberg) [inline src/avalanchemq/queue/queue.cr:539]:
```suggestion
          if expire_in <= 0
```

File: src/avalanchemq/queue/queue.cr [hunk 3]
```diff
@@ -536,19 +536,18 @@ module AvalancheMQ
...
+              i = 0
+            end
+            true
+          else
+            false
           end
-          true
         else
-          break
+          false
```
Comment (carlhoerberg) [inline src/avalanchemq/queue/queue.cr:550]: we won't break early now? so all messages in the ready queue will always be tested? or does returning false break?

File: src/avalanchemq/vhost.cr (hunk 1 above)
Comment (carlhoerberg) [reply src/avalanchemq/vhost.cr:43]: shouldn't `EXCHANGE_TYPES` then be in the http controller if it only affects the UI?

File: src/avalanchemq/queue/queue.cr (hunk 3 above)
Comment (snichme) [reply src/avalanchemq/queue/queue.cr:550]: `shift`  method in ReadyQueue breaks if block returns false

**Review by baelter — APPROVED** [review]

**Pull request merged**